packages/core/admin/admin/src/content-manager/components/RelationInput/RelationInput.js

```diff
@@ -143,6 +143,55 @@ const RelationInput = ({
     };
   }, [paginatedRelations, relations, numberOfRelationsToDisplay, totalNumberOfRelations]);
 
+  /**
+   * This code is being isolated because it's a hack to fix a placement bug in
+   * `react-select` where when the options prop is updated the position of the
+   * menu is not recalculated.
+   */
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const timeoutRef = useRef();
+
+  useEffect(() => {
+    setIsMenuOpen((isCurrentlyOpened) => {
+      /**
+       * If we're currently open and the options changed
+       * we want to close and open to ensure the menu's
+       * position is correctly calculated
+       */
+      if (isCurrentlyOpened) {
+        timeoutRef.current = setTimeout(() => {
+          setIsMenuOpen(true);
+        }, 10);
+
+        return false;
+      }
+
+      return false;
+    });
+  }, [options]);
+
+  useEffect(() => {
+    return () => {
+      /**
+       * If the component unmounts and a timer is set we should clear that timer
+       */
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+    };
+  }, []);
+
+  const handleMenuClose = (e) => {
+    setIsMenuOpen(false);
+    onSearchClose(e);
+  };
+
+  const handleMenuOpen = (e) => {
+    setIsMenuOpen(true);
+    onSearchOpen(e);
```

Comment:
after checking it seems we don't use anything from `e` in `onSearchOpen`
and weirdly we don't even use any function for `onSearchClose`, maybe it can be removed?

Reply:
Sure, good idea, I'll keep `onSearchClose` but i don't think it needs to be `required` so i've added an `if` check before we try call it.